### PR TITLE
Memory leak fixes

### PR DESCRIPTION
**Pull request by KTeofanidis**
Minor memory leak fixes

### CODE REVIEW

File: bftengine/include/bftengine/CryptoManager.hpp [hunk 1]
```diff
@@ -23,7 +23,9 @@ namespace bftEngine {
 class CryptoManager : public IKeyExchanger, public IMultiSigKeyGenerator {
  public:
   static CryptoManager& instance(const ReplicaConfig* config = nullptr, Cryptosystem* cryptoSys = nullptr) {
-    static CryptoManager cm_(config, cryptoSys);
+    // ensure cryptoSys will be freed since we're expected to take ownership
+    std::unique_ptr<Cryptosystem> cryptoSysPtr(cryptoSys);
```
Comment (teoparvanov) [inline bftengine/include/bftengine/CryptoManager.hpp:27]: I'm not sure why we need this intermediate `unique_ptr` here... In the `CryptoManager`, the raw pointer is reset into the `multiSigCryptoSystem_` unique_ptr. So I don't see how the raw pointer would leak, even without your changes?

Comment (KTeofanidis) [reply bftengine/include/bftengine/CryptoManager.hpp:27]: The CryptoManager construction is only called the first time. For subsequent calls the raw pointer is left dangling.

**Review by HristoStaykov — APPROVED** [review]
LGTM, thanks @KTeofanidis.

**Review by teoparvanov — APPROVED** [review]
LGTM @KTeofanidis , thanks for the clarifications!

**Pull request merged**